tests/test_core_network.py

```diff
@@ -500,12 +516,36 @@ def test_simplifing_puma_network_results_in_correct_record_of_removed_links_and_
     assert report['schedule']['schedule_level']['is_valid_schedule']
 
 
-def test_simplified_network_saves_to_correct_dtds(tmpdir, network_dtd, schedule_dtd):
-    n = read.read_matsim(path_to_network=puma_network_test_file, epsg='epsg:27700',
-                         path_to_schedule=puma_schedule_test_file)
+def test_simplify_does_not_oversimplify_PT_endpoints(
+        puma_network_pre_simplify, puma_network_post_simplify):
+    n = puma_network_pre_simplify
+
+    stops_at_risk = ['5221390681543854913', '5221390302070799085', '5221390323679791901']
+    for s in stops_at_risk:
+        assert n.link(n.schedule.stop(s).linkRefId)['length'] == 1
+
+    deleted_self_loops, n = puma_network_post_simplify
+
+    for s in stops_at_risk:
+        assert n.link(n.schedule.stop(s).linkRefId)['length'] == 1
+
+
+def test_simplify_removes_loops_by_default_but_keeps_pt_stop_loops(
+        puma_network_pre_simplify, puma_network_post_simplify):
+    n = puma_network_pre_simplify
```

Comment:
Again, the code under test is kind of hidden away in a fixture. Fixtures should just provide data to use for the test, really, they shouldn't be calling the code under test. They're part of the setup. We've also got some implicit knowledge of the network (particular link IDs) lurking in this test - is there a way to stop that knowledge from being spread around like this?

Reply:
I have to test saving a simplified network because the data types change (because we merge multiple links, they will have an average or max of a numeric attribute but for example osm tags will all be retained in a set or list and I need to be sure that that network is still valid under matsim. I would like to keep using a puma network that we simplify here because it's close to what we deal with in real life. I added something like this, is that a good compromise? If something goes wrong with simplification, it's caught and a message directs away from this test? 

<img width="1406" alt="Screenshot 2022-02-10 at 12 11 11" src="https://user-images.githubusercontent.com/36536946/153407394-ad5f37d8-689e-4d64-8f54-0090ab0987aa.png">

Reply:
changed to testing a dedicated fixture with data schema representative of a simplified network